packages/core/src/render3/view_ref.ts

```diff
@@ -284,7 +286,10 @@ export class ViewRef<T> implements viewEngine_EmbeddedViewRef<T>, viewEngine_Int
 
   attachToViewContainerRef() {
     if (this._appRef) {
-      throw new Error('This view is already attached directly to the ApplicationRef!');
+      const errorMessage = typeof ngDevMode === 'undefined' || ngDevMode ?
+          'This view is already attached directly to the ApplicationRef!' :
+          '';
+      throw new RuntimeError(RuntimeErrorCode.VIEW_ALREADY_ATTACHED, errorMessage);
```

Comment:
It should be safe to call `ngDevMode` without extra checks here as well:
```suggestion
      const errorMessage = ngDevMode ?
          'This view is already attached directly to the ApplicationRef!' :
          '';
      throw new RuntimeError(RuntimeErrorCode.VIEW_ALREADY_ATTACHED, errorMessage);
```